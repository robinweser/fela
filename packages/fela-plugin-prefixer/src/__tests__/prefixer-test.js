import prefixer from '../index'

describe('Prefixer plugin', () => {
  it('should prefix styles', () => {
    const style = {
      display: 'flex',
      justifyContent: 'center',
    }

    expect(prefixer()(style)).toMatchSnapshot()
  })

  it('should prefix nested objects', () => {
    const style = {
      display: 'flex',
      ':hover': {
        justifyContent: 'center',
      },
    }

    expect(prefixer()(style)).toMatchSnapshot()
  })

  it('handles css property values with line breaks correctly', () => {
    const style = {
      display: 'flex',
      backgroundImage: `linear-gradient(90deg,
      rgba(255,255,255, 0) 0,
      red 40%,
      red 60%,
      rgba(255,255,255, 0)
    )`,
      ':hover': {
        justifyContent: 'center',
      },
    }
    expect(prefixer()(style)).toMatchSnapshot()
  })
})

describe('Prefixer plugin (optimized)', () => {
  it('should prefix styles', () => {
    const plugin = prefixer().optimized

    expect(
      plugin({
        property: 'display',
        value: 'flex',
      })
    ).toEqual({
      property: 'display',
      value:
        '-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex',
    })

    expect(
      plugin({
        property: 'justifyContent',
        value: 'center',
      })
    ).toEqual({
      property: 'justifyContent',
      value: 'center',
    })
  })

  it('handles css property values with line breaks correctly', () => {
    const plugin = prefixer().optimized
    expect(
      plugin({
        property: 'backgroundImage',
        value: `linear-gradient(90deg,
      rgba(255,255,255, 0) 0,
      red 40%,
      red 60%,
      rgba(255,255,255, 0)
    )`,
      })
    ).toEqual({
      property: 'WebkitBackgroundImage',
      value: `linear-gradient(90deg,
      rgba(255,255,255, 0) 0,
      red 40%,
      red 60%,
      rgba(255,255,255, 0)
    );background-image:linear-gradient(90deg,
      rgba(255,255,255, 0) 0,
      red 40%,
      red 60%,
      rgba(255,255,255, 0)
    )`,
    })
  })
})
