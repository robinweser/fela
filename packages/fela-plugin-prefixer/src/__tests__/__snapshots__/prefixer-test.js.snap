// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prefixer plugin handles css property values with line breaks correctly 1`] = `
Object {
  ":hover": Object {
    "justifyContent": "center",
  },
  "WebkitBackgroundImage": "linear-gradient(90deg,
      rgba(255,255,255, 0) 0,
      red 40%,
      red 60%,
      rgba(255,255,255, 0)
    );background-image:linear-gradient(90deg,
      rgba(255,255,255, 0) 0,
      red 40%,
      red 60%,
      rgba(255,255,255, 0)
    )",
  "display": "-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex",
}
`;

exports[`Prefixer plugin should prefix nested objects 1`] = `
Object {
  ":hover": Object {
    "justifyContent": "center",
  },
  "display": "-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex",
}
`;

exports[`Prefixer plugin should prefix styles 1`] = `
Object {
  "display": "-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex",
  "justifyContent": "center",
}
`;
